lib/oli/resources/page_content.ex

```diff
@@ -69,9 +86,67 @@ defmodule Oli.Resources.PageContent do
   Maps the content elements of page content, preserving the as-is structure. Implemented as a
   convenience function, over top of map_reduce.
   """
-  def map(content, map_fn) do
-    {mapped, _} = map_reduce(content, 0, fn e, _ -> {map_fn.(e), 0} end)
+  def map(%{"model" => _model} = content, map_fn) do
+    {mapped, _} = map_reduce(content, 0, fn e, _acc, _tr_context -> {map_fn.(e), 0} end)
 
     mapped
   end
+
+  @doc """
+  Finds all activities contained in the given page content hierarchy and reduces a map of
+  activity resource ids to a map containing group and survey keys that indicate the
+  parent group and/or survey identifiers of the activity.
+
+  Useful for determining if an activity exists in a group or survey.
+
+  ## Examples
+      iex> activity_parent_groups(content)
+      %{
+        10 => %{group: "44325", survey: nil},        # inside a group
+        11 => %{group: nil, survey: "92374"},        # inside a survey
+        23 => %{group: "53222", survey: "96366"},    # inside a group and a survey
+        26 => %{group: nil, survey: nil},            # not inside a group or survey
+      }
+  """
+  def activity_parent_groups(%{"model" => _model} = content) do
```

Comment:
Can we move `activity_parent_groups` and `survey_activities` out of this module and into something that is more activity and survey specific?  It would be good to keep this module small and focused as an almost "Swiss army knife" collection of low level utilities that can be used to build task specific things like these two functions.